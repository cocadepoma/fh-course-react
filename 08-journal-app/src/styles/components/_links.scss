.link {
    text-decoration: none;
    color: $dark-grey;
    display: block;

    &:hover {
        text-decoration: underline;
    }
}
