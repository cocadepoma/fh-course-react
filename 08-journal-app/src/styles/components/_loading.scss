.loading__container {
    align-items: center;
    background-color: $primary;
    display: flex;
    justify-content: center;
    height: 100vh;
    width: 100vw;
}
.loading__container h1 {
    color: white;
}
