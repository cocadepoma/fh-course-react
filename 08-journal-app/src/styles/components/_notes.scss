.notes__main-content {
    display: flex;
    flex-direction: column;
    height: 100%;
}
.notes__appbar {
    align-items: center;
    background-color: $primary;
    color: white;
    display: flex;
    justify-content: space-between;
    padding: 10px 20px;
}
.notes__content {
    display: flex;
    flex-direction: column;
    height: 100%;
    padding: 20px;
}
.notes__title-input,
.notes__textarea {
    border: none;

    &:focus {
        outline: none;
    }
}

.notes__title-input {
    color: $dark-grey;
    font-size: 25px;
    font-weight: bold;
    margin-bottom: 10px;
}
.notes__textarea {
    border: none;
    color: $dark-grey;
    font-size: 20px;
    flex: 1 1 auto;
    resize: none;
    margin-bottom: 10px;
}
.notes__image img {
    -webkit-box-shadow: 5px 5px 5px 0px $dark-grey;
    -moz-box-shadow: 5px 5px 5px 0px $dark-grey;
    box-shadow: 5px 5px 5px 0px $dark-grey;
    max-width: 300px;
}
