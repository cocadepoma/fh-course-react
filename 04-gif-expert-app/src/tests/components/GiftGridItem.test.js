import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GiftGridItem } from "../../components/GiftGridItem";

describe("componente <GiftGridItem>", () => {
    const title = "gif de prueba";
    const url =
        "https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=05b050dfqw4732v9ots1n8gu46zle3j89wwcn76qa7mmwwh5&rid=giphy.gif";

    const wrapper = shallow(<GiftGridItem title={title} url={url} />);
    test("debe de mostrar el componente correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("debe de tener un pÃ¡rrafo con el title", () => {
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(title);
    });
    test("debe de tener la imagen igual al url y alt de los props", () => {
        const img = wrapper.find("img");
        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);
        // expect(img.props().src).toBe(url);
        // expect(img.props().alt).toBe(title);
    });

    test("debe de tener animate__fadeIn", () => {
        const div = wrapper.find("div");
        expect(div.hasClass("animate__fadeIn")).toBe(true);
    });
});
