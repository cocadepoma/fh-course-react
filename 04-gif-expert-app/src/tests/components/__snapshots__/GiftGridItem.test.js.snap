// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componente <GiftGridItem> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="gif de prueba"
    src="https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=05b050dfqw4732v9ots1n8gu46zle3j89wwcn76qa7mmwwh5&rid=giphy.gif"
  />
  <p>
    gif de prueba
  </p>
</div>
`;
